cilium-cli/install/helm.go

```diff
@@ -64,8 +64,12 @@ func (k *K8sInstaller) getHelmValues() (map[string]interface{}, error) {
 				// Can be removed once we drop support for <1.14.0
 				helmMapOpts["tunnel"] = tunnelDisabled
 			}
-			// TODO(tgraf) Is this really sane?
+			// AL2023 uses ens interfaces, but we default to eth interfaces for everything else for backwards compatibility,
+			// since the CLI was assuming eth interfaces before support for AL2023 was introduced
 			helmMapOpts["egressMasqueradeInterfaces"] = "eth0"
+			if k.params.AWS.AwsNodeImageFamily == AwsNodeImageFamilyAmazonLinux2023 {
```

Comment:
Do we know the interface naming pattern on all the image families listed (except custom)? Are all of them following eth0, except AL2023? Is it possible to retrieve the interface name directly from the host rather than infer it by the image family used?

The reason I ask is in the case where custom images based on AL2023 are used, or in the future other images follow the same convention.

Reply:
Neither the AWS CLI nor eksctl seem to expose this information, so it indeed would have to be retrieved from the host directly. There's the `ExecInPod` facility that could probably be used for that -- we can execute `ip address` commands and parse the output.

Reply:
we can leave this as a future enhancement. Thanks for checking.